FiveOhFirstDataCore/Pages/Personnel/Recruiter/AccountCreation.razor

```diff
@@ -211,17 +218,31 @@
 
         if((await _auth.AuthorizeAsync(claim, "RequireRecruiterOrReturningMemberStaff")).Succeeded)
         {
-            var res = await _roster.RegisterTrooper(TrooperData, claim);
-
-            if(res.GetResult(out var token, out var errors))
+            if (ValidNewId)
             {
-                _alert.PostAlert(this, $"New trooper added. Their Access Code is: {token}");
-                _advRefresh.CallDataReloadRequest("UserData");
-                TrooperData = new();
+                if (ValidNewNickname)
+                {
+                    var res = await _roster.RegisterTrooper(TrooperData, claim);
+
+                    if (res.GetResult(out var token, out var errors))
+                    {
+                        _alert.PostAlert(this, $"New trooper added. Their Access Code is: {token}");
+                        _advRefresh.CallDataReloadRequest("UserData");
+                        TrooperData = new();
+                    }
+                    else
+                    {
+                        Errors.AddRange(errors);
+                    }  
+                }
+                else
+                {
+                    _alert.PostAlert(this, $"Trooper name is invalid");
```

Comment:
The overload for `_alert.PostAlert(this, "a string");` is for success messages. The overload for `_alert.PostAlert(this, new List<string> { "a string" });` is for error messages. There is a third overload that takes a alert config class for further customization.

Reply:
Corrected